syntax = "proto3";

message GetConfigRes {
  int32 offset = 1;
  uint32 time = 2;
}

message GetConfigReq {
  int32 offset = 1;
  uint32 time = 2;
  int32 lock_password = 3;
  int32 lock_time = 4;
  int32 limit_power_mypower = 5;
  int32 zero_export_433_addr = 6;
  int32 zero_export_enable = 7;
  int32 netmode_select = 8;
  int32 channel_select = 9;
  int32 server_send_time = 10;
  int32 wifi_rssi = 11;
  int32 serverport = 12;
  bytes apn_set = 13;
  bytes meter_kind = 14;
  bytes meter_interface = 15;
  bytes wifi_ssid = 16;
  bytes wifi_passward = 17;
  bytes server_domain_name = 18;
  int32 inv_type = 19;
  bytes dtu_sn = 20;
  int32 access_model = 21;
  int32 mac_0 = 22;
  int32 mac_1 = 23;
  int32 mac_2 = 24;
  int32 mac_3 = 25;
  int32 dhcp_switch = 26;
  int32 ip_addr_0 = 27;
  int32 ip_addr_1 = 28;
  int32 ip_addr_2 = 29;
  int32 ip_addr_3 = 30;
  int32 subnet_mask_0 = 31;
  int32 subnet_mask_1 = 32;
  int32 subnet_mask_2 = 33;
  int32 subnet_mask_3 = 34;
  int32 default_gateway_0 = 35;
  int32 default_gateway_1 = 36;
  int32 default_gateway_2 = 37;
  int32 default_gateway_3 = 38;
  bytes ka_nub = 39;
  bytes apn_name = 40;
  bytes apn_passward = 41;
  int32 sub1g_sweep_switch = 42;
  int32 sub1g_work_channel = 43;
  int32 cable_dns_0 = 44;
  int32 cable_dns_1 = 45;
  int32 cable_dns_2 = 46;
  int32 cable_dns_3 = 47;
  int32 wifi_ip_addr_0 = 48;
  int32 wifi_ip_addr_1 = 49;
  int32 wifi_ip_addr_2 = 50;
  int32 wifi_ip_addr_3 = 51;
  int32 mac_4 = 52;
  int32 mac_5 = 53;
  int32 wifi_mac_0 = 54;
  int32 wifi_mac_1 = 55;
  int32 wifi_mac_2 = 56;
  int32 wifi_mac_3 = 57;
  int32 wifi_mac_4 = 58;
  int32 wifi_mac_5 = 59;
  bytes gprs_imei = 60;
  bytes dtu_ap_ssid = 61;
  bytes dtu_ap_pass = 62;
}
